api/src/main/java/com/epam/pipeline/security/saml/SAMLUserDetailsServiceImpl.java

```diff
@@ -79,7 +83,7 @@ public UserContext loadUserBySAML(SAMLCredential credential) {
         String userName = credential.getNameID().getValue().toUpperCase();
         List<String> groups = readAuthorities(credential);
         Map<String, String> attributes = readAttributes(credential);
-
+        final UserContext userContext;
```

Comment:
It is a good time to clean up this code a bit.
I suggest to use `Optional` and replace `if` in the following manner:
```
Optional.ofNullable(userManager.loadUserByName(userName))
.map(user -> processRegisteredUser(user, ...))
.orElseGet(() -> processNewUser(...))
```